packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/EmitterRpc/Emitter.cs

```diff
@@ -16,11 +16,11 @@ public sealed class Emitter : IDisposable
         private static Emitter? _emitter;
         private bool _disposed;
 
-        private readonly StreamWriter _writer;
+        private readonly TextWriter _writer;
 
-        private Emitter()
+        internal Emitter()
         {
-            _writer = new StreamWriter(Console.OpenStandardOutput()) { AutoFlush = true };
+            _writer = Console.Out;
```

Comment:
`Console.OpenStandardOutput()` always returns the stream corresponding to the console's output which is not switchable - therefore I changed to this so that we could change the stream to a `StringWriter` to test it.